presto-main/src/main/java/com/facebook/presto/execution/TaskExecutor.java

```diff
@@ -855,6 +888,20 @@ public long getRunningTasksLevel4()
     }
 
     @Managed
+    public long getLongestSplitTimeMillis()
+    {
+        Iterator<SplitProfile> iterator = splitProfiler.iterator();
```

Comment:
JIT doesn't plan for exceptions, so this can really trip it over. Since this is profiling code, I'd say it's important that we don't use exceptions unless absolutely necessary.